### ISSUE TITLE
TS error when declaring "pattern" using P.string as a variable.

### ISSUE THREAD

**Issue by momentiris** [gvergnaud/ts-pattern#174]
Hello and thanks for the great work on this library!

I'm currently bumping TS and ts-pattern to v5 and ran into an unexpected issue.
Prior to v5, I've had some patterns declared to variables, like this:
```ts
const isFoo = {
  type: 'foo',
  data: P.string
}

match(unknown)
  .with(isFoo, (foo) => ...)
  .otherwise(() => ...)
```

After upgrading ts and ts-pattern, I now get an error on `isFoo`:

`The inferred type of this node exceeds the maximum length the compiler will serialize. An explicit type annotation is needed. (tsserver 7056)`

Removing the `P.string` also removes the error
Using `P.string` "inline" does not give this error.

ts-pattern __5.0.1__
TS __5.1.6__

Any ideas?

Thanks in advance.

**Comment by gvergnaud** [comment]
Hey, could you provide a playground with a reproduction case? I do not reproduce this issue: [Playground](https://www.typescriptlang.org/play?#code/JYWwDg9gTgLgBAbziAhjAxgCwDRwApwC+cAZlBCHAOQwDOAtGGjAKZQB2VA3AFA-oR2teMFoAxCBDgBeRDzhwYATzAsAXNRKSq2eXAAmaFBrwA6YVGDsA5j0J99LdABsUUFnAFD4ADw0BXdgBrdggAd3Y+VAxMAAofAEo9UzDgGDjRCQhcWK0IBJkAPjgAIhKkhVMIdLZU2hZY2ILpYrKEoA)

**Comment by momentiris** [comment]
That's strange. I realised now that the error actually arises when you export `isFoo`. I didn't include that in my initial example, but it's something I do in the project where I'm facing the issue.

[playground with export](https://www.typescriptlang.org/play?#code/JYWwDg9gTgLgBAbziAhjAxgCwDRwApwC+cAZlBCHAOQwDOAtGGjAKZQB2VA3AFA8sAPSLDjoI7WvGC0AYhAhwAvIh5w4MAJ5gWALmol5VbKrgATNCj14AdJKjB2Acx6E+plugA2KKC1HjJOAE9AFd2AGt2CAB3dj5UDEwACgEAShNraOAYZOk5CFwkgwhUpQA+OAAiSvS1awgctizaFiSk0sUK6tSgA)

**Comment by gvergnaud** [comment]
Thanks to the help of the amazing @Andarist, I think I have a fix: https://github.com/gvergnaud/ts-pattern/pull/175

I'll release this tomorrow

**Comment by momentiris** [comment]
Look promising! Thanks for the quick assistance!

**Comment by gvergnaud** [comment]
Fixed in v5.0.3
